proxy/src/ctx/transport.rs

```diff
@@ -27,6 +28,19 @@ pub struct Client {
     pub proxy: Arc<ctx::Proxy>,
     pub remote: SocketAddr,
     pub metadata: destination::Metadata,
+    pub tls_status: TlsStatus,
+}
+
+/// Identifies whether or not a connection was secured with TLS,
+/// and the reason why not.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum TlsStatus {
```

Comment:
My first inclination is that this belongs somewhere else, not in ctx.  This is used to configure                                 connections, so does it belong in `destination`? Or does it belong in the transport module?

Reply:
Actually, let me clarify: If this reflects "how was this configured", it belongs in ctx. If this is used ever to determine "how should this be configured", it should not.

Reply:
This is used to reflect how the connection was configured, not to configure it.